src/importer.rs

```diff
@@ -416,4 +454,248 @@ mod tests {
+        cache.set(path.clone(), metadata, &models);
+
+        let loaded_models = cache.get_by_checksum(checksum);
+
+        assert_eq!(loaded_models.expect("Models should be loaded"), models);
+    }
+
+    #[test]
+    fn test_obj_cache_get_by_checksum_returns_none_if_checksum_does_not_match() {
+        let mut cache = EndlessCache::default();
+        let path = "/path/to/some.obj".to_string();
+        let checksum = 1u32;
+        let metadata = FileMetadata {
+            checksum,
+            last_modified: SystemTime::now(),
+        };
+        let models = tobj_to_internal(vec![create_tobj_model(vec![], vec![], vec![])]);
+        cache.set(path.clone(), metadata, &models);
+
+        let loaded_models = cache.get_by_checksum(checksum + 1);
+
+        assert!(loaded_models.is_none());
+    }
+
+    // The following tests with mocked cache are technically integration tests
+    // and they use fixture data. They're kept here to prevent complications
+    // with automocks not being present in debug build, as it is built without
+    // code marked as `cfg(test)`.
+    //
+    // FIXME: Ideal scenario would be if filesystem was mocked and passed into
+    // this method as well. It'd become proper unit test that way.
+
+    #[test]
+    fn test_importer_import_obj_cache_sets_models_if_file_was_not_cached_before() {
+        let mut cache = MockObjCache::new();
+        cache
+            .expect_get_if_not_modified()
+            .returning(|_, _| None)
+            .times(1);
+        cache.expect_get_by_checksum().returning(|_| None).times(1);
+        cache.expect_set().returning(|_, _, _| ()).times(1);
+
+        let mut importer = Importer::new(cache);
+        let path = "tests/fixtures/valid.obj";
+
+        importer
+            .import_obj(&path)
+            .expect("Valid obj should be loaded");
+    }
+
+    #[test]
+    fn test_importer_import_obj_cache_returns_unmodified_file() {
+        let mut cache = MockObjCache::default();
+        cache
+            .expect_get_if_not_modified()
+            .returning(|_, _| Some(vec![]))
+            .times(1);
+        cache.expect_get_by_checksum().returning(|_| None).times(0);
+        cache.expect_set().returning(|_, _, _| ()).times(0);
+
+        let mut importer = Importer::new(cache);
+        let path = "tests/fixtures/valid.obj";
+
+        importer
+            .import_obj(&path)
+            .expect("Valid obj should be loaded");
+    }
+
+    #[test]
+    fn test_importer_import_obj_cache_returns_the_same_cached_data_from_different_file() {
+        let mut cache = MockObjCache::new();
+        cache
+            .expect_get_if_not_modified()
+            .returning(|_, _| None)
+            .times(1);
+        cache
+            .expect_get_by_checksum()
+            .returning(|_| Some(vec![]))
+            .times(1);
+        cache.expect_set().returning(|_, _, _| ()).times(1);
```

Comment:
Is there a way to assert we are given the same data here as we produced in `get_by_checksum`? I believe `mockall` has a `.with()` thing that sounds like it may help.

Reply:
Yes, there is and will try to do that. I thought this lazy attempt would be enough, but you're right that we should test that as well.